Makefile

```diff
@@ -357,5 +357,13 @@ clean-examples: ## Remove all the temporary files generated in the examples fold
 	rm -f examples/provider-components/provider-components-*.yaml
 
 .PHONY: verify
-verify: ## Runs verification scripts to ensure correct execution
+verify: verify-boilerplate verify-modules
+
+.PHONY: verify-boilerplate
+verify-boilerplate:
 	./hack/verify-boilerplate.sh
+
+.PHONY: verify-modules
+verify-modules: modules
+	git diff-index --quiet HEAD -- go.sum go.mod hack/tools/go.mod hack/tools/go.sum
```

Comment:
I'm curious why you chose `git diff-index` instead of `git diff`?

Reply:
`git diff-index` shows only changed files, `git diff` shows full diffs. Though, both fails silently if `--quiet` flag is specified, so they both suits our use-case.

Reply:
Do you think we should produce some error text that indicates what failed? Otherwise it just looks like this (I pretended to bump the controller-runtime version):
```
$ make verify-modules
go mod tidy
go: finding sigs.k8s.io/controller-runtime v0.2.1
go: downloading sigs.k8s.io/controller-runtime v0.2.1
go: extracting sigs.k8s.io/controller-runtime v0.2.1
cd hack/tools; go mod tidy
git diff-index --quiet HEAD -- go.sum go.mod hack/tools/go.mod hack/tools/go.sum
make: *** [verify-modules] Error 1
```